src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output
+} from '@angular/core';
+
+import {
+  SkyInlineFormCloseArgs
+} from './';
+
+@Component({
+  selector: 'sky-inline-form',
+  templateUrl: './inline-form.component.html',
+  styleUrls: ['./inline-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyInlineFormComponent {
+
+  @Output()
+  public closed = new EventEmitter<SkyInlineFormCloseArgs>();
+
+  @Input()
+  public showDeleteButton = false;
+
+  public onDoneClick() {
+    this.closeWithReason('save');
+  }
+
+  public onDeleteClick() {
+    this.closeWithReason('delete');
+  }
+
+  public onCancelClick() {
+    this.closeWithReason('cancel');
+  }
+
+  private closeWithReason(reason: string) {
+    const args: SkyInlineFormCloseArgs = {
+      reason: reason
```

Comment:
Can be simplified as:
```
{
  reason
}
```

Reply:
The code changed a bit for the customizable buttons. I think this is moot now.